src/node_process_methods.cc

```diff
@@ -454,6 +425,83 @@ static void ReallyExit(const FunctionCallbackInfo<Value>& args) {
   env->Exit(code);
 }
 
+class FastHrtime {
+ public:
+  explicit FastHrtime(std::shared_ptr<v8::BackingStore> backing_store)
+      : backing_store_(backing_store) {}
+
+  static Local<Object> New(Environment* env) {
+    auto create_func = [env](auto fast_func, auto slow_func) {
+      auto cfunc = v8::CFunction::Make(fast_func);
+      return v8::FunctionTemplate::New(env->isolate(),
+                                       slow_func,
+                                       Local<Value>(),
+                                       Local<v8::Signature>(),
+                                       0,
+                                       v8::ConstructorBehavior::kThrow,
+                                       v8::SideEffectType::kHasNoSideEffect,
+                                       &cfunc);
+    };
+
+    auto number_tmpl = create_func(FastNumber, SlowNumber);
+    auto bigint_tmpl = create_func(FastBigInt, SlowBigInt);
+
+    Local<v8::ObjectTemplate> otmpl = v8::ObjectTemplate::New(env->isolate());
+    otmpl->SetInternalFieldCount(1);
+    otmpl->Set(FIXED_ONE_BYTE_STRING(env->isolate(), "hrtime"), number_tmpl);
+    otmpl->Set(FIXED_ONE_BYTE_STRING(env->isolate(), "hrtimeBigInt"),
+               bigint_tmpl);
+    Local<Object> obj = otmpl->NewInstance(env->context()).ToLocalChecked();
+
+    Local<ArrayBuffer> ab = ArrayBuffer::New(env->isolate(), 16);
+    FastHrtime* data = new FastHrtime(ab->GetBackingStore());
+    obj->SetAlignedPointerInInternalField(0, data);
+    obj->Set(
+           env->context(), FIXED_ONE_BYTE_STRING(env->isolate(), "buffer"), ab)
+        .ToChecked();
+
+    return obj;
+  }
+
+ private:
+  // This is the legacy version of hrtime before BigInt was introduced in
+  // JavaScript.
+  // The value returned by uv_hrtime() is a 64-bit int representing nanoseconds,
+  // so this function instead fills in an Uint32Array with 3 entries,
+  // to avoid any integer overflow possibility.
+  // The first two entries contain the second part of the value
+  // broken into the upper/lower 32 bits to be converted back in JS,
+  // because there is no Uint64Array in JS.
+  // The third entry contains the remaining nanosecond part of the value.
+  static void FastNumber(FastHrtime* receiver) {
+    uint64_t t = uv_hrtime();
+    uint32_t* fields = static_cast<uint32_t*>(receiver->backing_store_->Data());
+    fields[0] = (t / NANOS_PER_SEC) >> 32;
+    fields[1] = (t / NANOS_PER_SEC) & 0xffffffff;
+    fields[2] = t % NANOS_PER_SEC;
```

Comment:
Might be able to make this a bit nicer using the `AliasedStruct` utility

Reply:
I'm not sure, we're really just looking at the raw view. AliasedStruct is more about representing a single `T` using an ArrayBuffer right?

Reply:
Yep. Feel free to ignore this if it's not useful ;-) ... absolutely love that you're working on this